Sources/Operator/abstract_operator.hpp

```diff
@@ -100,16 +100,16 @@ class AbstractOperator {
   Member function returning the hilbert space associated with this Hamiltonian.
   @return Hilbert space specifier for this Hamiltonian
   */
-  const AbstractHilbert &GetHilbert() const { return *hilbert_; }
+  std::shared_ptr<const AbstractHilbert> GetHilbert() const { return hilbert_; }
 
```

Comment:
Why don't you add a constructor taking a shared_ptr to an `hilbert` here, so that we can avoid having to call `SetHilbert` at every construction of derived classes but we can simply call the base constructor?

Reply:
Agreed